src/components/SaveSegmentGroupDialog.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <v-card>
+    <v-card-title class="d-flex flex-row align-center">
+      Save Segment Group
+    </v-card-title>
+    <v-card-text>
+      <v-form v-model="valid" @submit.prevent="saveSegmentGroup">
+        <v-text-field
+          v-model="fileName"
+          hint="Filename that will appear in downloads."
+          label="Filename"
+          :rules="[validFileName]"
+          required
+          id="filename"
+        />
+
+        <v-select
+          label="Format"
+          v-model="fileFormat"
+          :items="EXTENSIONS"
+        ></v-select>
+      </v-form>
+    </v-card-text>
+    <v-card-actions>
+      <v-spacer />
+      <v-btn
+        :loading="saving"
+        color="secondary"
+        @click="saveSegmentGroup"
+        :disabled="!valid"
+      >
+        <v-icon class="mr-2">mdi-content-save</v-icon>
+        <span data-testid="save-confirm-button">Save</span>
+      </v-btn>
+    </v-card-actions>
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { onKeyDown } from '@vueuse/core';
+import { saveAs } from 'file-saver';
+import { useSegmentGroupStore } from '@/src/store/segmentGroups';
+import { writeImage } from '@/src/io/readWriteImage';
+import { useErrorMessage } from '@/src/composables/useErrorMessage';
+
+const EXTENSIONS = [
+  'nrrd',
+  'nii',
+  'nii.gz',
+  'dcm',
+  'hdf5',
+  'tif',
+  'mha',
+  'vtk',
+  'iwi.cbor',
+];
+
+const props = defineProps<{
+  close: () => void;
```

Comment:
Any reason for passing close as a prop vs emitting a close event? Just to be clear, this is fine as-is, but vue tends to prefer emitting events.

Reply:
Good tip.  React instincts. I changed it to an event.